src/Altinn.App.Core/Helpers/Serialization/ModelSerializationService.cs

```diff
@@ -0,0 +1,267 @@
+using System.Collections.Concurrent;
+using System.Text;
+using System.Text.Json;
+using System.Xml;
+using System.Xml.Serialization;
+using Altinn.App.Core.Features;
+using Altinn.App.Core.Internal.AppModel;
+using Altinn.Platform.Storage.Interface.Models;
+
+namespace Altinn.App.Core.Helpers.Serialization;
+
+/// <summary>
+/// DI registered service for centralizing (de)serialization logic for data models
+/// </summary>
+public class ModelSerializationService
```

Comment:
Since this is public, is it worth hiding it behind an interface? Makes it a little more obvious that it should be retrieved from DI. Should we also make the namespace `Altinn.App.Core.Features.Serialization` or something? I think we talked about centralizing new stuff there in the future

Reply:
Yes, With the addition of `(de)SerialiseFromStorage` and possibly `(de)SerializeFromRequest` it kind of grew from a helper to a service.